Engine/src/Engine/EngineUI/UserInterface.h

```diff
@@ -45,14 +49,19 @@ namespace Engine
 		//partition defines the title section of the UI element
 		void partition(const std::string& fontTag, const std::string& title = "", const ImVec4& color = white);
 
-		virtual void show() = 0;
+		virtual void update() {}
+		virtual void show(unsigned int fbo = 0) = 0;
```

Comment:
do all children of UserInterface expect and fbo to passed in? What if the panel has no rendering involved?

Reply:
You're right, the others did not need to be passed fbo. It was there for testing but I forgot to remove it